mleh! (Outputs unhelpful error message)

I mistyped the name of the cabal file in my CI setup, and I got this error message:

```
get-tested: mleh!
CallStack (from HasCallStack):
  error, called at app/Main.hs:28:15 in main:Main
Error: Process completed with exit code 1.
```

This is fun, but not helpful ;) a better error message would have been something like `mleh! File foo.cabal could not be found!`

--- a/app/Extract.hs
+++ b/app/Extract.hs
@@ -19,6 +19,7 @@ import Effectful
 import Effectful.Error.Static (Error, throwError)
 import System.Directory qualified as System
 import System.IO (stdout)
+
 import Types
 
 -- | Loads and parses a Cabal file

--- a/app/Main.hs
+++ b/app/Main.hs
@@ -7,8 +7,10 @@ import Data.Function ((&))
 import Data.Vector qualified as Vector
 import Effectful
 import Effectful.Error.Static
-import Extract
 import Options.Applicative
+import System.Exit
+
+import Extract
 import Types
 
 data Options = Options
@@ -25,7 +27,12 @@ main = do
   processingResult <- runEff . runErrorNoCallStack $ runOptions result
   case processingResult of
     Right json -> ByteString.putStrLn $ "matrix=" <> json
-    Left _ -> error "mleh!"
+    Left (CabalFileNotFound path) -> do
+      putStrLn $ "Could not find cabal file at path " <> path
+      exitFailure
+    Left (CabalFileCouldNotBeParsed path) -> do
+      putStrLn $ "Could not parse cabal file at path " <> path
+      exitFailure
 
 parseOptions :: Parser Options
 parseOptions =

